environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/NetworkCreationValidator.java

```diff
@@ -32,39 +34,80 @@ public NetworkCreationValidator(NetworkService networkService) {
 
     public ValidationResultBuilder validateNetworkCreation(Environment environment, NetworkDto network, Map<String, CloudSubnet> subnetMetas) {
         ValidationResultBuilder resultBuilder = new ValidationResultBuilder();
-        if (AWS.name().equalsIgnoreCase(environment.getCloudPlatform()) && network != null && network.getNetworkCidr() == null) {
-            String message;
-            if (network.getSubnetIds().size() < 2) {
-                message = "Cannot create environment, there should be at least two subnets in the network";
-                LOGGER.info(message);
-                resultBuilder.error(message);
+        if (network != null && network.getNetworkCidr() == null) {
+            if (AWS.name().equalsIgnoreCase(environment.getCloudPlatform())) {
```

Comment:
Instead of ifing here, EnvironmentNetworkProviderValidator should decide which provider specific validator should be used